### PR TITLE
Add the 2201.2.0 release note to B.io

### PR DESCRIPTION
**Pull request by praneesha**
## Purpose
Add the 2201.2.0 release note to B.io.
> Fixes #

## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

**Review by sm1990 — APPROVED** [review]

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Review by mohanvive — APPROVED** [review]

**Review by dulajdilshan — APPROVED** [review]

**Review by daneshk — APPROVED** [review]

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md [hunk 1]
```diff
@@ -0,0 +1,759 @@
...
+    };
+
+    table<Employee> key(id) employees = table [
+        // Results in an error now.
+        {...employee}
+    ];
+}
+```
+
+To view bug fixes, see the [GitHub milestone for 2201.2.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3AType%2FBug+label%3ATeam%2FCompilerFE+milestone%3A%22Ballerina+2201.2.0%22).
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:410]: Is this link correct? @gimantha / @praneesha ?

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md [hunk 2]
```diff
@@ -0,0 +1,759 @@
...
+A call to `onGracefulStop` will result in one call to the handler function that was passed as an argument; the handler functions will be called after calling `gracefulStop` on all registered listeners, in the reverse order of the corresponding calls to `onGracefulStop`.
+
+E.g., a `foo` function can be called during the graceful shutdown by registering it as follows.
+
+`runtime:onGracefulStop(foo);`
+
+#### Improvements
+
+
+#### Type-reference type support at runtime
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:437]: This should be a L5 heading ? @praneesha

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md (hunk 1 above)
Comment (anupama-pathirage) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:410]: It should be https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3AType%2FBug+label%3ATeam%2FCompilerFE+milestone%3A%222201.2.0%22

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md [hunk 3]
```diff
@@ -0,0 +1,759 @@
...
+To view bug fixes, see the [GitHub milestone for 2201.2.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-standard-library/issues?q=is%3Aclosed+is%3Aissue+milestone%3A%222201.2.0%22+label%3AType%2FBug).
+
+### Code to Cloud updates
+
+#### Improvements
+The base image was updated to `ballerina/jvm-runtime:1.0` based on Alpine 3.15 with the necessary libraries.
+
+#### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.2.0 (Swan Lake)](https://github.com/ballerina-platform/module-ballerina-c2c/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22Ballerina+2201.2.0%22+label%3AType%2FBug).
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:581]: Seems it has no bugs. So can't we remove the bug fixes section? @xlight05

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md [hunk 4]
```diff
@@ -0,0 +1,759 @@
...
+        // Edges
+        "org/package":"0.1.0" -> "ballerina/io":"1.2.2";
+}
+```
+
+##### Compiler API
+
+- Introduced a set of builders in the Types API, which are used to construct complex types that have varying components
+
+##### Ballerina Shell
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:661]: Empty section? @mohanvive / @praneesha ?

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md [hunk 5]
```diff
@@ -0,0 +1,759 @@
...
+
+##### Ballerina Update Tool
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:663]: Empty section @keizer619 / @praneesha ?

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md (hunk 4 above)
Comment (mohanvive) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:661]: @praneesha let's remove this topic from this.

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md [hunk 6]
```diff
@@ -0,0 +1,759 @@
...
+- Added completion support for the resource access action context
+
+##### OpenAPI Tool
+Added support for OpenAPI schema constraint properties in client/service generation. With this improvement, the OpenAPI constraints will be applied as `ballerina/constraint` standard library package annotations when generating Ballerina clients and services from the OpenAPI definition.
+The following OpenAPI properties are currently supported in the Ballerina OpenAPI generation tool.
+- `minimum`, `maximum`, `exclusiveMinimum`, and `exclusiveMaximum` for `integer` and `number` types
+- `minLength` and `maxLength` for `string` type
+- `minItems` and `maxItems` for `array` type
+
+To view bug fixes, see the GitHub milestone for 2201.2.0 (Swan Lake) of the repositories below.
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:678]: `#### Bug fixes` is missing

Comment (anupama-pathirage) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:678]: Seems both urls are not having any issues. So can we remove the links @mohanvive / @keizer619 ?

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md [hunk 7]
```diff
@@ -0,0 +1,759 @@
...
+};
+```
+
+>**Note:** The entire JSON value to Ballerina record conversion logic is rewritten to enhance the conversion experience. The above-mentioned scenarios are a few noticeable, important changes.
+
+##### Compiler API
+
+- Added semantic API support for the client resource access action
+
+##### Ballerina shell
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:737]: Empty title @keizer619

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md [hunk 8]
```diff
@@ -0,0 +1,759 @@
...
+
+##### Ballerina update Tool
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:739]: Empty titlle @keizer619

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md [hunk 9]
```diff
@@ -0,0 +1,759 @@
...
+
+#### Bug fixes
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:741]: Empty title ? @praneesha ?

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md [hunk 10]
```diff
@@ -0,0 +1,759 @@
...
+
+### Breaking changes
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:743]: Empty title ? @praneesha ?

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md (hunk 3 above)
Comment (xlight05) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:581]: Theres one, I missed the milestone. Added now.

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md (hunk 2 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:437]: Fixed.

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md (hunk 4 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:661]: Removed.

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md (hunk 1 above)
Comment (gimantha) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:410]: @anupama-pathirage Yes.. at this moment, the commented url is the correct one. But aren't we changing the milestones once released? All other Previous milestones are prefixed with "Ballerina"

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md (hunk 5 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:663]: Removed.

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md (hunk 1 above)
Comment (gimantha) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:410]: @NipunaMadhushan Any thoughts?

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md (hunk 9 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:741]: Removed.

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md (hunk 10 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:743]: Removed.

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md (hunk 8 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:739]: Removed.

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md (hunk 6 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:678]: Removed.

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md (hunk 1 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:410]: @dulajdilshan @keizer619 - Please confirm if the link is correct.

Comment (dulajdilshan) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md]: The commented link shows the correct issues tagged for the milestone. Are we changing the milestone after the release to Ballerina 2201.2.0?

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md (hunk 6 above)
Comment (mohanvive) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md:676]: @praneesha It seems above url for Language Server is wrong. Below is the correct one, please add it, https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3ATeam%2FLanguageServer+milestone%3A2201.2.0

File: downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md (hunk 1 above)
Comment (keizer619) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md]: We are not changing milestones as we release. I think we can keep just the without "Ballerina" part

Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md]: Updated to https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3AType%2FBug+label%3ATeam%2FCompilerFE+milestone%3A%222201.2.0%22.

**Review by anupama-pathirage — APPROVED** [review]

**Review by xlight05 — APPROVED** [review]

**Pull request merged**